core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/Utils.kt

```diff
@@ -337,28 +337,110 @@ internal fun <T : Comparable<T>> T.between(left: T, right: T, includeBoundaries:
         this > left && this < right
     }
 
-private const val DELIMITERS = "[_\\s]"
-public val DELIMITERS_REGEX: Regex = DELIMITERS.toRegex()
-public val DELIMITED_STRING_REGEX: Regex = ".+$DELIMITERS.+".toRegex()
+// Single regex to split words by non-alphanumeric characters, camelCase, and numbers
 
-internal val CAMEL_REGEX = "(?<=[a-zA-Z])[A-Z]".toRegex()
+internal val CAMEL_DEFAULT_DELIMITERS_REGEX =
+    "[^a-zA-Z0-9]+|(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])|(?<=\\d)(?=[a-zA-Z])|(?<=[a-zA-Z])(?=\\d)".toRegex()
 
-public fun String.toCamelCaseByDelimiters(delimiters: Regex): String = split(delimiters).joinToCamelCaseString()
+/**
+ * Converts a string into lowerCamelCase using [delimiters].
+ *
+ * - Splits words using a unified regex that handles delimiters.
```

Comment:
Maybe specify shortly what this means, something like "Splits words on any characters that are not letters or digits" (that's true right?), it's clearer that way.